--- Mercurial.ini	Thu Apr 30 12:20:42 2009
+++ /z/mozilla-build/Mercurial.ini	Thu Apr 30 12:28:05 2009
@@ -3,8 +3,6 @@
 ; USERNAME is your Windows user name:
 ;   C:\Documents and Settings\USERNAME\Mercurial.ini
 
-[ui]
-editor = notepad
 ; show changed files and be a bit more verbose if True
 ; verbose = True
 
@@ -114,3 +112,13 @@
 winmerge.regname=Executable
 winmerge.checkchanged=True
 winmerge.gui=True
+
+[diff]
+git = 1
+showfunc = 1
+
+[extensions]
+hgext.graphlog =
+
+[defaults]
+diff =-p -U 8
