--- Mercurial.rc.orig	2011-08-01 19:17:08 -0700
+++ Mercurial.rc	2011-08-16 20:41:17 -0700
@@ -16,7 +16,7 @@
 
 [ui]
 ; editor used to enter commit logs, etc.  Most text editors will work.
-editor = notepad
+; editor = notepad
 ; show changed files and be a bit more verbose if True
 ; verbose = True
  
@@ -46,13 +46,13 @@ editor = notepad
 ;extdiff =
 ;fetch =
 ;gpg =
-;graphlog = 
+graphlog = 
 ;hgcia =
 ;hgk =
 ;highlight = 
 ;interhg =
 ;keyword =
-;mq =
+mq =
 ;notify =
 ;pager =
 ;patchbomb =
@@ -64,6 +64,9 @@ editor = notepad
 ;win32mbcs =
 ;zeroconf =
 
+[defaults]
+diff =-p -U 8
+
 ;
 ; Define external diff commands
 ;
@@ -88,6 +91,7 @@ editor = notepad
 ; expect normal patches. so enable git patches at your own risk.
 ;
 [diff]
-;git = false
+git = true
+showfunc = true
 ;nodates = false
 
