--- Mercurial.rc.old	2013-02-20 20:47:06 -0500
+++ Mercurial.rc	2013-02-21 08:42:19 -0500
@@ -16,7 +16,6 @@
 
 [ui]
 ; editor used to enter commit logs, etc.  Most text editors will work.
-editor = notepad
 ; show changed files and be a bit more verbose if True
 ; verbose = True
  
@@ -39,13 +38,13 @@
 ;bugzilla =
 ;children =
 ;churn =
-;color =
+color =
 ;convert =
 ;eol =
 ;extdiff =
 ;fetch =
 ;gpg =
-;graphlog = 
+graphlog = 
 ;hgcia =
 ;hgk =
 ;highlight = 
@@ -53,21 +52,35 @@
 ;interhg =
 ;largefiles =
 ;keyword =
-;mq =
+mq =
 ;notify =
-;pager =
+pager =
 ;patchbomb =
 ;progress =
-;purge =
-;rebase =
+purge =
+rebase =
 ;record =
 ;relink =
 ;schemes =
 ;share =
-;transplant =
+transplant =
 ;win32mbcs =
 ;zeroconf =
 
+[defaults]
+diff =-p -U 8
+
+[color]
+mode = ansi
+
+[pager]
+pager = less.exe -mnFRSXQ
+quiet = True
+attend = diff,status,log,qdiff,blame,annotate,pdiff,glog,show,qseries
+
+[alias]
+show = log --patch --verbose --rev
+
 ;
 ; Define external diff commands
 ;
@@ -92,6 +105,7 @@
 ; expect normal patches. so enable git patches at your own risk.
 ;
 [diff]
-;git = false
+git = true
+showfunc = true
 ;nodates = false
 
