--- Mercurial.rc	2014-07-15 23:25:00 -0400
+++ Mercurial.rc	2014-07-15 23:25:25 -0400
@@ -11,17 +11,16 @@
 ; your user Mercurial.ini file, where USERNAME is your Windows user name:
 ;
 ; XP or older    - C:\Documents and Settings\USERNAME\Mercurial.ini
 ; Vista or later - C:\Users\USERNAME\Mercurial.ini
 
 
 [ui]
 ; editor used to enter commit logs, etc.  Most text editors will work.
-editor = notepad
 ; show changed files and be a bit more verbose if True
 ; verbose = True
  
 ; username data to appear in commits
 ; it usually takes the form: Joe User <joe.user@host.com>
 ; username = Joe User <j.user@example.com>
 
 ; In order to push/pull over ssh you must specify an ssh tool
@@ -35,45 +34,59 @@ editor = notepad
 ; Extensions shipped with Mercurial
 ;
 [extensions]
 ;acl =
 ;blackbox =
 ;bugzilla =
 ;children =
 ;churn =
-;color =
+color =
 ;convert =
 ;eol =
 ;extdiff =
 ;fetch =
 ;gpg =
 ;hgcia =
 ;hgk =
 ;highlight = 
 ;histedit =
 ;interhg =
 ;largefiles =
 ;keyword =
-;mq =
+mq =
 ;notify =
-;pager =
+pager =
 ;patchbomb =
 ;progress =
-;purge =
-;rebase =
+purge =
+rebase =
 ;record =
 ;relink =
 ;schemes =
 ;share =
 ;shelve =
-;transplant =
+transplant =
 ;win32mbcs =
 ;zeroconf =
 
+[defaults]
+diff =-p -U 8
+
+[color]
+mode = ansi
+
+[pager]
+pager = less.exe -mnFRSXQ
+quiet = True
+attend = diff,status,log,qdiff,blame,annotate,pdiff,glog,show,qseries
+
+[alias]
+show = log --patch --verbose --rev
+
 ;
 ; Define external diff commands
 ;
 [extdiff]
 ;cmd.bc3diff = C:\Program Files\Beyond Compare 3\BCompare.exe
 ;cmd.vdiff = C:\Progra~1\TortoiseSVN\bin\TortoiseMerge.exe
 ;cmd.vimdiff = gvim.exe
 ;opts.vimdiff = -f "+next" "+execute 'DirDiff' fnameescape(argv(0)) fnameescape(argv(1))"
@@ -88,11 +101,12 @@ editor = notepad
 
 ;
 ; The git extended diff format can represent binary files, file
 ; permission changes, and rename information that the normal patch format
 ; cannot describe.  However it is also not compatible with tools which
 ; expect normal patches. so enable git patches at your own risk.
 ;
 [diff]
-;git = false
+git = true
+showfunc = true
 ;nodates = false
 
